packages/thirdweb/src/analytics/track.ts

```diff
@@ -20,10 +20,38 @@ export function trackConnect(args: {
   });
 }
 
+/**
+ * @internal
+ */
+export function trackPayEvent(args: {
+  client: ThirdwebClient;
+  event: string;
+  walletAddress?: string;
+  fromToken?: string;
+  fromAmount?: string;
+  toToken?: string;
+  toAmount?: string;
+  chainId?: number;
+  dstChainId?: number;
+}) {
+  track(args.client, {
+    source: "pay",
+    action: args.event,
```

Comment:
few things i'd like to have here:

- payMode (can be in the action) - top-up | commerce | transaction
- walletType

Reply:
Going to track the pay mode in the initial event (when the modal first renders) unless you think we should have it in all events (would have to pass the prop around)

Reply:
for the pay mode, i think we actually already have it in most places via the PayOptions that we pass around no?